### ISSUE TITLE
Issue with TypeScript

### ISSUE THREAD

**Issue by betoharres** [Sharcoux/rn-css#16]
Hello, I got stuck trying to make my first component, maybe it's because I'm not experienced with typescript.
While trying to setup my first component, I stumbled into this issue:
```Javascript
import {Theme} from '~/app/styles';
import styled from 'rn-css';

export const Container = styled.View`
  ${({shared}: {shared: Theme}) => `
    flex: 1;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    background-color: ${shared.colors.Black};
  `}
`;
```
I got a compile error on `styled.View` statement, saying:
```
Property 'View' does not exist on type '<T>(Component: ComponentType<T>) => <S>(chunks: TemplateStringsArray, ...functs: (string | number | boolean | ((arg: S & T & { rnCSS?: string | undefined; shared: unknown; }) => string | ... 3 more ... | undefined) | null | undefined)[]) => ForwardRefExoticComponent<...>'.
```
Maybe the message is missing something since I set up typescript on Vim, which looks it works better on VScode.
I got into the source code of this library and I was able to fix the error by changing this on my code:
```
import {Theme} from '~/app/styles';
import {View} from 'react-native';
import styled from 'rn-css';

export const Container = styled(View)`
  ${({shared}: {shared: Theme}) => `
    flex: 1;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    background-color: ${shared.colors.Black};
  `}
`;
```
so instead of accessing View as a property of styled, I tried to pass the View component as a argument and it worked, the error is gone and the screen also displayed just fine as before(on both ways they rendered OK).

Any idea that I did something wrong or was it something from the library? Thanks!!

**Comment by Sharcoux** [comment]
This seems very weird. I never had anything like that. Let me investigate.

**Comment by Sharcoux** [comment]
`styled.View` is just an alias for `styled(View)`, so don't worry, it will behave as expected anyway.

Ok. My hypothesis:

1) Your version of rn-css is wrong
2) Your version of typescript is wrong
3) The SharedValue context is generating a cyclic dependency or some weird behaviour (we don't use it in our codebase, so it's not much tested)

But  checking the code:

![image](https://user-images.githubusercontent.com/797545/135365134-d47bed74-deb2-4580-a6fd-b90a32b30792.png)

I don't see how styled can be defined, but not styled.View...

**Comment by Sharcoux** [comment]
What typescript version are you using?

**Comment by betoharres** [comment]
Hello @Sharcoux , thanks for the fast reply and sorry my late response!
You were right, I added the latest typescript version and it did stop the errors being thrown(I was using TS v3.9.10). Actually, when I first created the project I tried adding the latest TS version, but it kept throwing a warning every time I ran `eslint`, like so:
```
00:47:41 ~/code/iov $ yarn lint
yarn run v1.22.11
$ eslint . --ext .js,.jsx,.ts,.tsx
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.1.0

YOUR TYPESCRIPT VERSION: 4.4.3

Please only submit bug reports when using the officially supported version.

=============
```
so I kept the version(v3.9.10) the TS template installed when I ran:
``npx react-native init MyApp --template react-native-template-typescript``
which is the command recommended in the react-native docs to initialize a new TS project.
Anyway, looks like it's all good, I'm going to keep going with your library on my new project!
Thanks again for the hard work!
